### PR TITLE
[addons] Sort logged list of installed add-ons

### LINKED ISSUE #14825: Please order the installed addon list alphabetically
### Describe the bug
When Kodi Leia starts, a list of installed addons is printed on the log like this
```
 Running the application...
19:07:30.467 T:7188  NOTICE: ES: Starting UDP Event server on port 9777
19:07:30.467 T:7188  NOTICE: UDP: Listening on port 9777 (ipv6 : false)
19:07:32.747 T:8028  NOTICE: Skin Helper Service --> WebService - start helper webservice on port 52307
19:07:32.747 T:4120  NOTICE: Skin Helper Service --> Skin Helper Service version 1.1.3 started
19:07:41.621 T:6128  NOTICE: ADDON: resource.images.moviegenreicons.transparent v0.0.6 installed
19:07:41.621 T:6128  NOTICE: ADDON: game.libretro.beetle-bsnes v0.9.26.1 installed
19:07:41.621 T:6128  NOTICE: ADDON: xbmc.json v9.7.2 installed
19:07:41.621 T:6128  NOTICE: ADDON: metadata.common.fanart.tv v3.5.0 installed
19:07:41.621 T:6128  NOTICE: ADDON: script.module.autocompletion v1.1.2 installed
19:07:41.621 T:6128  NOTICE: ADDON: game.controller.ps.guncon.japan v1.0.8 installed
...
```

## Expected Behavior
It would be very nice if this list is printed in alphabetical order to quickly find a particular addon and version.

## Actual Behavior
The addon list is in a random order.

### To Reproduce
Install Kodi Leiba beta 5, start Kodi, see kodi.log

### Your Environment
Used Operating system: tested in Windows 7 64 bit with Kodi Leia Beta 5

### PR DESCRIPTION
**Pull request by yol**
## Description
Fixes #14825

## Motivation and Context
Difficult to find specific add-ons in list otherwise

## How Has This Been Tested?
Build/run Linux x86_64

### CODE REVIEW

**Comment by chewitt** [comment]
Tested and working fine with LE - thanks for doing this!

File: xbmc/addons/AddonManager.cpp [hunk 1]
```diff
@@ -688,17 +691,28 @@ bool CAddonMgr::FindAddons()
...
+      {
+        CLog::Log(LOGNOTICE, "ADDON: {} v{} installed", installedAddon.first, installedAddon.second);
+      }
+
+      std::set<std::string> installedIdentifiers;
+      std::transform(
+          installed.cbegin(), installed.cend(),
+          std::inserter(installedIdentifiers, installedIdentifiers.begin()),
+          [](const decltype(installed)::value_type& p) { return p.first; }
+      );
```
Comment (Rechi) [inline xbmc/addons/AddonManager.cpp]: I would add to both sets in the for loop instead of transorming later.

**Comment by pkerling** [comment]
Had that idea too, but tbh I think it's 99% the same here. Will change if you insist though.

**Comment by AlwinEsch** [comment]
Is not the most beautiful way, but to have sorted the list is much better than before, for me fully OK.

**Review by Rechi — APPROVED** [review]

**Pull request merged**